### PR TITLE
[Qt6Base] Upgrade to OpenSSL 3

### LINKED ISSUE #7531: Bump Qt6Base to use OpenSSL 3
Are there any plans to do so?

### PR DESCRIPTION
**Pull request by simonbyrne**
I'm not exactly sure if this is the correct fix?

Fixes #7531.

cc: @barche

### CODE REVIEW

**Comment by simonbyrne** [comment]
Okay, well we're a couple of versions behind (6.5.3 and 6.6.0 are both available: https://download.qt.io/archive/qt/), so should we just update those?

**Comment by giordano** [comment]
Yup, that'd do it.

File: Q/Qt6Base/build_tarballs.jl [hunk 1]
```diff
@@ -176,7 +176,7 @@ dependencies = [
     Dependency("Glib_jll", v"2.59.0"; compat="2.59.0"),
     Dependency("Zlib_jll"),
     Dependency("CompilerSupportLibraries_jll"),
-    Dependency("OpenSSL_jll"; compat="1.1.10"),
+    Dependency("OpenSSL_jll"; compat="3"),
```
Comment (giordano) [inline Q/Qt6Base/build_tarballs.jl:179]:
```suggestion
    Dependency("OpenSSL_jll"; compat="3.0.8"),
```

**Comment by simonbyrne** [comment]
Okay, no idea if this is the right way to do this. @barche does this look right?

**Comment by simonbyrne** [comment]
okay, i'm out of ideas...

**Comment by barche** [comment]
I’m away from my computers until Sunday, but 6.5.3 is the right version to choose, going to 6.6 would require rebuilding GR.

**Comment by barche** [comment]
Looks like the new version requires some extra patching to deal with the AVX stuff.

**Comment by simonbyrne** [comment]
thanks @barche!

**Review by giordano — APPROVED** [review]

**Comment by yuyichao** [comment]
I assume it's unintentional that this removes everything but x64 linux musl? https://github.com/JuliaBinaryWrappers/Qt6Base_jll.jl/commit/382b0ad96f67ebe1c55dc92a921fbb48d76a5a3b

**Comment by yuyichao** [comment]
OK I saw https://github.com/JuliaPackaging/Yggdrasil/pull/7628#issuecomment-1793832399

**Pull request merged**